tests/components/downloader/test_services.py

```diff
@@ -0,0 +1,69 @@
+"""Test downloader action."""
+
+from http import HTTPStatus
+from unittest.mock import mock_open, patch
+
+import pytest
+import requests_mock
+
+from homeassistant.components.downloader.const import (
+    ATTR_AUTH_PASSWORD,
+    ATTR_AUTH_TYPE,
+    ATTR_AUTH_USERNAME,
+    ATTR_URL,
+    DOMAIN,
+    SERVICE_DOWNLOAD_FILE,
+)
+from homeassistant.const import HTTP_DIGEST_AUTHENTICATION
+from homeassistant.core import HomeAssistant
+
+
+def digest_challenge_matcher(request):
+    """Match requests with no auth header."""
+    return "Authorization" not in request.headers
+
+
+def digest_response_matcher(request):
+    """Match requests with an auth header."""
+    return "Authorization" in request.headers
+
+
+@pytest.mark.usefixtures("loaded_config_entry")
+async def test_digest_auth(hass: HomeAssistant) -> None:
```

Comment:
I think it would be nicer with the fixture, then you can un-indent most of the code below:
```suggestion
async def test_digest_auth(hass: HomeAssistant, requests_mock: Mocker) -> None:
```